llvm/lib/CodeGen/MachinePassManager.cpp

```diff
@@ -90,7 +90,9 @@ Error MachineFunctionPassManager::run(Module &M,
       for (unsigned I = Begin, E = Idx; I != E; ++I) {
         auto *P = Passes[I].get();
 
-        if (!PI.runBeforePass<MachineFunction>(*P, MF))
+        // Keep BeforeStack empty in ChangeReporter
+        if (P->name() != FreeMachineFunctionPass::name() &&
```

Comment:
we should invoke `runAfterPassInvalidated` below if we see that we're running `FreeMachineFunctionPass`, rather than not running instrumentation on it

Reply:
I have something that does this in my rewrite

Reply:
we can keep this for now, but https://github.com/llvm/llvm-project/pull/81068 will overwrite this